sdk/storage/storage-queue/src/utils/utils.common.ts

```diff
@@ -371,23 +371,33 @@ export function sanitizeHeaders(originalHeader: HttpHeaders): HttpHeaders {
  * @returns {string} with the account name
  */
 export function getAccountNameFromUrl(url: string): string {
-  if (url.startsWith("http://127.0.0.1:10000")) {
-    // Dev Conn String
-    return getValueInConnString(DevelopmentConnectionString, "AccountName");
-  } else {
-    try {
+  const parsedUrl: URLBuilder = URLBuilder.parse(url);
+  let accountName;
+  try {
+    if (parsedUrl.getHost()!.split(".")[1] === "queue") {
       // `${defaultEndpointsProtocol}://${accountName}.queue.${endpointSuffix}`;
       // Slicing off '/' at the end if exists
       url = url.endsWith("/") ? url.slice(0, -1) : url;
 
-      const accountName = url.substring(url.lastIndexOf("://") + 3, url.lastIndexOf(".queue."));
-      if (!accountName) {
-        throw new Error("Provided accountName is invalid.");
+      accountName = parsedUrl.getHost()!.split(".")[0];
+    } else {
+      if (url.startsWith("http://127.0.0.1:10000")) {
```

Comment:
http://127.0.0.1:10000 is not a sepcial case of IP style url parsing, why put it into a special if block?

Reply:
You're right, I was thinking of the shorthand notation of connection string without a proper endpoint to parse... but then it would be caught even before and will be replaced with the proper endpoint!

Reply:
Fixed!